### PR TITLE
Python 3 + CKAN 2.9 + Github Actions

### PR DESCRIPTION
**Pull request by nickumia-reisys**
Only creating to try and spur github actions into ... action?

### CODE REVIEW

**Comment by nickumia-reisys** [comment]
Sorry! This was meant to be a PR into our own fork...

Btw, these tests are to demonstrate that the extension works in CKAN 2.8 and 2.9 😅

**Comment by nickumia-reisys** [comment]
After talking with my team, we wanted to open this up to see if upstream would like the upgraded tests running in CKAN 2.9.  The main extension requires no update, but the tests were upgraded to run with pytests.  Please let me know if there's any questions!

**Comment by amercader** [comment]
We would love some 2.9 tests and GitHub Actions! 

We don't want to keep the nose tests and there should be just a `test.py` file (no `tests_2_8_and_above.py`). To make sure the same test suite will work across different CKAN versions add a `dev-requirements.txt` file with `pytest-ckan` as a requirement (like [here](https://github.com/ckan/ckanext-dcat/blob/master/dev-requirements.txt#L3)) and install them in the workflow. That will install all requirements even in CKAN<2.9.

This PR also includes a lot of files that seem to be project specific. We shouldn't need the `.env`, `docker-compose.yml`, `Makefile` etc files. Github Actions should pull all the images needed, you can check an existing extension for how the test workflow is defined:

https://github.com/ckan/ckanext-dcat/blob/master/.github/workflows/test.yml

**Comment by amercader** [comment]
> To make sure the same test suite will work across different CKAN versions add a dev-requirements.txt file with pytest-ckan as a requirement (like here) and install them in the workflow.

I see now that you do this [here](https://github.com/okfn/ckanext-envvars/pull/4/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R57). It's still best to put it on `dev-requirements.txt` so other developers can use it locally

**Comment by nickumia-reisys** [comment]
@amercader Sorry for the unnaturally long wait, but I updated this PR to address your comments from above.  Please let me know if there's more work to do!

The Data.gov team was hoping that okfn would create another PyPI release with this new code as well 🚀

**Comment by nickumia-reisys** [comment]
## Successful Github Actions
image

File: README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    pip install -r dev-requirements.txt
+
+Running the Tests
+-----------------
+
+They follow the guidelines for [testing CKAN extensions](https://docs.ckan.org/en/2.8/extensions/testing-extensions.html#testing-extensions).
+
+To run the tests, do:
+
+> pytest \--ckan-ini=test.ini ckanext/dcat/tests
```
Comment (amercader) [inline README.md]:
```suggestion
> pytest --ckan-ini=test.ini ckanext/envvars/tests.py
```

File: ckanext/envvars/tests.py [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
-import ckan
 import ckan.plugins.toolkit as toolkit
 import ckan.plugins as p
 
+try:
+    p.toolkit.requires_ckan_version("2.8")
+except p.toolkit.CkanVersionException:
+    from pylons import config
+else:
+    from ckan.common import config
```
Comment (amercader) [inline ckanext/envvars/tests.py]: For plugins that need to work across different CKAN versions is best to use the [ckantoolkit](https://github.com/ckan/ckantoolkit) package, which wraps toolkit objects to make them available in different CKAN versions. Add it as a requirement to `setup.py` ( using `install_requires=["ckantoolkit"]`, you can leave it unpinned)

```suggestion
from ckantoolkit import config
```

You can replace all other usage of the toolkit with

```python

import ckantoolkit as toolkit
```

File: setup.py [hunk 3]
```diff
@@ -44,6 +44,7 @@
         # that you indicate whether you support Python 2, Python 3 or both.
         'Programming Language :: Python :: 2.6',
```
Comment (amercader) [inline setup.py]: You can remove 2.6

File: test.ini [hunk 4]
```diff
@@ -9,7 +9,7 @@ host = 0.0.0.0
 port = 5000
 
 [app:main]
-use = config:../ckan/test-core.ini
```
Comment (amercader) [inline test.ini]: Don't change this path, it's the standard on a source install, instead update the path in the GA workflow like in here:

https://github.com/ckan/ckanext-geoview/blob/9ddc29cfcdb66c20109eef82eb06b04c65fdb69b/.github/workflows/test.yml#L53

**Review by amercader — COMMENTED** [review]
This looks great @nickumia-reisys !
Just some minor comments and we are good to go

**Comment by nickumia-reisys** [comment]
Thanks for the thorough review @amercader!  All of your suggestions were well explained and help to get it back to the CKAN standard.  I hope I didn't miss anything 🎯 

When this does get merged, don't forget to enabled Github Actions for the repo (if they're not already enabled) and the Data.gov team would greatly appreciate if there was a new PyPI release 😄 

If it is of any interest to the CKAN team, there is an automated Github Action that publishes packages to PyPI (the only setup required is adding the PyPI token to the Github Secrets).  Here's an example: https://github.com/GSA/ckanext-envvars/blob/main/.github/workflows/deploy.yml

**Comment by amercader** [comment]
@nickumia-reisys thanks, ckanext-envvars==0.0.2 is available on pypi :rocket:

**Pull request merged**